File: testers/avro/kotlin/generated-and-checked-in/com/example/events/consumer/CustomerOrderConsumer.kt
package com.example.events.consumer

import com.example.events.CustomerOrder
import com.example.events.header.StandardHeaders
import java.io.Closeable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords

/** Type-safe consumer for customer-order topic */
data class CustomerOrderConsumer(
  val consumer: Consumer<kotlin.String, CustomerOrder>,
  val handler: CustomerOrderHandler,
  val topic: kotlin.String = "customer-order"
) : Closeable {
  /** Close the consumer */
  override fun close() {
    consumer.close()
  }

  /** Poll for messages and dispatch to handler */
  fun poll(timeout: Duration) {
    val records: ConsumerRecords<kotlin.String, CustomerOrder> = consumer.poll(timeout)
    records.forEach({ record -> val key: kotlin.String = record.key()
    val value: CustomerOrder = record.value()
    val headers: StandardHeaders = StandardHeaders.fromHeaders(record.headers())
    handler.handle(key, value, headers) })
  }
}

File: testers/avro/scala/generated-and-checked-in/com/example/events/consumer/CustomerOrderConsumer.scala
package com.example.events.consumer

import com.example.events.CustomerOrder
import com.example.events.header.StandardHeaders
import java.lang.AutoCloseable
import java.time.Duration
import org.apache.kafka.clients.consumer.Consumer
import org.apache.kafka.clients.consumer.ConsumerRecords

/** Type-safe consumer for customer-order topic */
case class CustomerOrderConsumer(
  consumer: Consumer[String, CustomerOrder],
  handler: CustomerOrderHandler,
  topic: String = "customer-order"
) extends AutoCloseable {
  /** Poll for messages and dispatch to handler */
  def poll(timeout: Duration): Unit = {
    val records: ConsumerRecords[String, CustomerOrder] = consumer.poll(timeout)
    records.forEach(record => { val key: String = record.key; val value: CustomerOrder = record.value; val headers: StandardHeaders = StandardHeaders.fromHeaders(record.headers); handler.handle(key, value, headers) })
  }

  /** Close the consumer */
  override def close: Unit = {
    consumer.close
  }
}